packages/babel-plugin-transform-destructuring/src/util.ts

```diff
@@ -620,6 +617,8 @@ export function convertVariableDeclaration(
     }
   }
 
+  const mustVarOrExpr = t.isForStatement(path.parent) && path.key !== "body";
```

Comment:
```suggestion
  const inForInit = t.isForStatement(path.parent, { init: node })
```

I slightly prefer shorter variable name.